internal/cache/file/downloader/jm_parallel_downloads_test.go

```diff
@@ -145,9 +138,9 @@ func TestParallelDownloads(t *testing.T) {
 				select {
 				case jobStatus := <-subscriberC:
 					if assert.Nil(t, err) {
-						assert.Equal(t, tc.expectedOffset, jobStatus.Offset)
+						assert.GreaterOrEqual(t, tc.objectSize, jobStatus.Offset)
 						verifyFileTillOffset(t,
-							data.FileSpec{Path: util.GetDownloadPath(path.Join(cacheDir, storage.TestBucketName), "path/in/gcs/foo.txt"), FilePerm: util.DefaultFilePerm, DirPerm: util.DefaultDirPerm}, tc.expectedOffset,
+							data.FileSpec{Path: util.GetDownloadPath(path.Join(cacheDir, storage.TestBucketName), "path/in/gcs/foo.txt"), FilePerm: util.DefaultFilePerm, DirPerm: util.DefaultDirPerm}, jobStatus.Offset,
```

Comment:
We should probably verifyFileTillOffset iff assert.GreaterOrEqual has returned true i.e.

```
if assert.Nil(t, err) && assert.GreaterOrEqual(t, tc.objectSize, jobStatus.Offset) {
    verifyFileTillOffset(...)
}
```

Reply:
When we have multiple asserts, 2nd assert gets executed only if 1st assert succeeds. so its the same.